example/WORKSPACE

```diff
@@ -41,24 +42,72 @@ load("@rules_haskell//haskell:cabal.bzl", "stack_snapshot")
 # Haskell dependencies and toolchain
 ######################################
 
+load("@io_tweag_gazelle_cabal//:config_settings/setup.bzl", "config_settings")
+
+config_settings(name = "config_settings")
+
+load("@config_settings//:info.bzl", "ghc_version")
 load("@io_tweag_gazelle_cabal//:defs.bzl", "gazelle_cabal_dependencies")
 
 gazelle_cabal_dependencies()
 
 stack_snapshot(
     name = "stackage",
+    components =
+        {
+            "tasty-discover": [
+                "lib",
+                "exe:tasty-discover",
+            ],
+        } if ghc_version == "8.10.7" else {
+            "tasty-discover": [
+                "lib",
+                "exe:tasty-discover",
+            ],
+            "attoparsec": [
+                "lib",
+                "lib:attoparsec-internal",
+            ],
+        },
+    components_dependencies =
+        None if ghc_version == "8.10.7" else {
+            "attoparsec": """{"lib:attoparsec": ["lib:attoparsec-internal"]}""",
+        },
+    local_snapshot = "@io_tweag_gazelle_cabal//:snapshot-" + ghc_version + ".yaml",
     packages = [
+        "Cabal",  #keep
+        "base",
+        "inspection-testing",
         "json",  # keep
+        "mtl",
         "path",  # keep
         "path-io",  #keep
+        "tasty",
+        "tasty-discover",
+        "tasty-hunit",
+        "text",
+        "void",
```

Comment:
Perhaps we could let the dependencies not marked with `# keep` to be discovered by `gazelle update-repos` instead of listing them explicitly.